staging/src/k8s.io/apimachinery/pkg/runtime/serializer/codec_factory.go

```diff
@@ -69,11 +70,19 @@ func newSerializersForScheme(scheme *runtime.Scheme, mf json.MetaFactory, option
 			json.SerializerOptions{Yaml: false, Pretty: true, Strict: options.Strict},
 		)
 	}
-
```

Comment:
Something about this function and the slice of serializerTypes it returns doesn't sit well with me.

1. We don't have a serializer for pretty && strict, every option we add is going to balloon the number of serializerTypes exponentially.
2. We don't seem to use options.Strict and options.Pretty consistently. We have a separate serializer for pretty, but just pass the strict options through.
3. Having both a serializer that is always strict and a serializer that is variably strict based on `options` smells repugnant.

I have a hunch that we could do something about the ballooning constructors in [negotiation/negotiate.go](https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/negotiation/negotiate.go#L69-L92) that would make our lives easier here, but I haven't fully thought it through yet.

That might be too much yak shaving for the validation task at hand, there's likely a way simpler way I'm not thinking of.

Reply:
This is (should) only be used for DEserialization, it may seem less bad if you name it accordingly? Pretty only makes sense when writing and strict only makes sense when reading.

... the next time we add something we'll probably have this problem though.